packages/ra-core/src/controller/ListController.js

```diff
@@ -166,8 +169,15 @@ export class ListController extends Component {
 
     setPage = page => this.changeParams({ type: SET_PAGE, payload: page });
 
-    setFilters = filters =>
-        this.changeParams({ type: SET_FILTER, payload: filters });
+    setFilters = debounce(filters => {
+        if (isEqual(filters, this.props.filterValues)) {
```

Comment:
I'm not sure about this check. In the previous version, we kept a local version of the `filters`, and now you use `filterValues`

Are you sure this is correct?

Reply:
It works when running demo. Yet, E2E tests fail. I'm checking out why.

Reply:
let me ask this in a different way: why did you remove the local cache with `this.filters`?